cmd/legacy_main.go

```diff
@@ -297,19 +297,7 @@ func runCLIApp(c *cli.Context) (err error) {
 	// if these are already being logged into a log-file, otherwise
 	// there will be duplicate logs for these in both places (stdout and log-file).
 	if newConfig.Foreground || newConfig.Logging.FilePath == "" {
-		flagsStringified, err := util.Stringify(*flags)
-		if err != nil {
-			logger.Warnf("failed to stringify cli flags: %v", err)
-		} else {
-			logger.Infof("GCSFuse mount command flags: %s", flagsStringified)
-		}
-
-		mountConfigStringified, err := util.Stringify(*mountConfig)
-		if err != nil {
-			logger.Warnf("failed to stringify config-file: %v", err)
-		} else {
-			logger.Infof("GCSFuse mount config flags: %s", mountConfigStringified)
-		}
+		logger.Logger().Info("GCSFuse config", "config", newConfig)
```

Comment:
Can you confirm the log printed by this is the exact same as it was generated by util.Stringify ?
The util.Stringify supported returning error earlier, but I see that `logger.Logger().Info("GCSFuse config", "config", newConfig)` doesn't . What happens to the error cases now ?

Reply:
Error gets printed in the log. It'll look something like:
{"timestamp":{"seconds":1723196744,"nanos":102771135},"severity":"INFO","message":"GCSFuse config","config":"<error_msg>"}

Reply:
> It won't be exactly the same.

Could you share the older log and newer both so we could see the difference ? I am concerned that if we change the format enough, it might affect those who are dependent on our existing log format.

> Error gets printed in the log. It'll look something like:

I see problems with that change. It changes what used to be a `[WARNING]` log earlier, and would now be an info/debug log, which is easier to miss than warning log. I don't know exactly where the stringify was supposed to fail, but if was not a harmless failure, then I think we should keep it as warning/error log.

Reply:
@kislaykishore ^

Reply:
@gargnitingoogle Who is dependent on our existing log format?

Reply:
Regarding the error scenario: we are using mostly basic types and the custom types implement `MarshalText()` which don't return error. So, no errors will be thrown while logging and if in the very remote case they are thrown, they will still get logged as INFO. The way that we are logging now changes the format and makes it a lot cleaner. So, I think it's worth doing it considering the trade-off involved.

Reply:
> Who is dependent on our existing log format?

GCSFuse logs are a publicly feature, available to all our users and are especially used by internal users and are also used by us for debugging, so changing the log format should not be silently done.

I suggest that we call out the format change in our documentation (not sure what the best place is).

> we are using mostly basic types and the custom types implement MarshalText() which don't return error. So, no errors will be thrown while logging

According to https://pkg.go.dev/encoding/json#Marshal, the error could have occurred for one of the following: floating point value NaN, Channel, complex, and function values, cyclic data structures, so theoretically it's possible to have an error if we're missing a check on the values of any floating point in the config, which is hard to guarantee in the future.

> if in the very remote case they are thrown, they will still get logged as INFO

Is it okay to change it to the following ?

```go
if _, err := json.Marshal(config); err != nil {
    logger.Logger().Warning("GCSFuse config", "config", newConfig)
} else {
    logger.Logger().Info("GCSFuse config", "config", newConfig)
}
```

or some variation of it.

This will keep logging at INFO if there is no issue, and will log a warning if there is an error.

Reply:
Since it's used for debugging, I'll apprise the team regarding the log format change.

We are keeping things simple. So, such cases don't really apply here and I don't want to add further complexity if it's not needed and that too when we are not losing on the error.

The code you shared ties the logging format with json which is probably not a good idea.

Reply:
@gargnitingoogle Could you review this again? I wouldn't want to increase code complexity due to cases that are not gonna happen. Seems like a straightforward application of [YAGNI](https://martinfowler.com/bliki/Yagni.html) and [KISS](https://pattersonjunior.medium.com/the-kiss-principle-cb480c9adcf8) principles.

Reply:
Sure.